testing/test_recursive_dict.py

```diff
@@ -0,0 +1,123 @@
+from neuraxle.hyperparams.space import RecursiveDict, HyperparameterSamples
+
+
+def test_recursive_dict_to_flat():
+    dict_values = {
+        'hp': 1,
+        'stepa': {
+            'hp': 2,
+            'stepb': {
+                'hp': 3
+            }
+        }
+    }
+    r = RecursiveDict(separator='__', **dict_values)
```

Comment:
we should have a test for different separators when nesting. 

Question: what happens when a separator changes from one level to another? 
Answer: I think each level could have a different separator, and the separator of a level is the one just to the right of the first child in the getitem.

Reply:
To test what I mean: 

```
dict 1: separator="__"
dict 2: separator="."
dict 3: separator="---"
dict 4: separator="__"
```

test getting this: "some__hp.in---some__step"
That means 4 nested dict with especially the 4th one with the same param as the 1st one. So the argument is in the 5th.